token/token.go

```diff
@@ -308,12 +309,16 @@ func LookupIdent(ident string) *Token {
 // only have one possible instance/value
 // (ie all the tokens except for the first 4 value tokens).
 // TODO: codegen all the token constants to avoid needing this function.
-// (even though that's better than string comparaisons).
+// (even though that's better than string comparisons).
 func ByType(t Type) *Token {
 	return tToT[t]
 }
 
 func (t *Token) Literal() string {
+	if t == nil {
+		log.Critf("Nil token .Literal() called")
+		return "NIL_TOKEN"
```

Comment:
Is there a risk if someone write NIL_TOKEN and ask grol to parse it ?

Reply:
no it's if you lookup a token in a map and get nil and then call this method (in this case it was @ being looked up and not found in the map and then still used to print error, I fixed that but left the log for the next time a bug like that shows up (though the natural panic NPE might be fine too)